src/site/content/en/blog/codelab-payment-form-best-practices/index.md

```diff
@@ -0,0 +1,340 @@
+to simulate mobile devices. 
+* [Send the URL from your computer to your phone](https://support.google.com/chrome/answer/9430554).
+* Use a service such as [BrowserStack](https://www.browserstack.com/open-source) to test on a range 
+of devices and browsers. 
+
+## Step 3: Add attributes to help users enter data
+
+Enable the browser to store and autofill input values, and provide access to secure built-in 
+payment and validation features.
+
+Add attributes to the form in your `index.html` file so it looks like this:
+
+```html/5,6,10,11,16,17,20,21
+<form action="#" method="post">
+
+  <h1>Payment form</h1>
+
+  <section>        
+    <label for="cc-number">Card number</label>
+    <input id="cc-number" name="cc-number" autocomplete="cc-number" inputmode="numeric" pattern="[\d ]{10,30}" required>
+  </section>
+
+  <section>        
+    <label for="cc-name">Name on card</label>
+    <input id="cc-name" name="cc-name" autocomplete="cc-name" pattern="[\p{L} \-\.]+" required>
+  </section>
+  
+  <section id="cc-exp-csc">      
+    <div>
+      <label for="cc-exp">Expiry date</label>
+      <input id="cc-exp" name="cc-exp" autocomplete="cc-exp" placeholder="MM/YY" maxlength="5" required>
+    </div> 
+    <div>
+      <label for="cc-csc">Security code</label>
+      <input id="cc-csc" name="cc-csc" autocomplete="cc-csc" inputmode="numeric" maxlength="3" required>
+      <div class="explanation">Back of card, last 3 digits</div>
+    </div>
+  </section>  
+
+  <button id="complete-payment">Complete payment</button>
+
+</form>
+```
+
+View your app again and then tap or click in the **Card number** field. Depending on the device and 
+platform, you may see a chooser showing payment methods stored for the browser, like the one below. 
+
+Once you select a payment method and enter your security code, the browser autofills the form using 
+the payment card `autocomplete` values you added to the form:
+
+* `cc-number`
+* `cc-name`
+* `cc-exp`
+* `cc-csc`
+
+Many browsers also check and confirm the validity of credit card numbers and security codes.
+
+{% Aside %}
+Don't be alarmed! Your browser may be able to autofill the form with stored payment card data, but 
+no payment can be made, and no data is transferred or saved.
+{% endAside %}
+
+On a mobile device you'll also notice that you get a numeric keyboard as soon as you tap into the 
+**Card number** field. That's because you used `inputmode="numeric"`. For numeric fields this makes 
+it easier to enter numbers and impossible to enter non-numeric characters, and nudges users to 
+remember the type of data they're entering.
+
+<figure class="w-figure">
+  <img 
+    src="images/credit-card-autofill.png" 
+    alt="Two screenshots of a payment form in Chrome on an Android phone. One shows the built-in 
+    browser payment card selector; the other shows dummy autofilled values.">
+  <figcaption class="w-figcaption">Built-in browser payment chooser and autofill.</figcaption>
+</figure>
+
```

Comment:
```suggestion
View your app again and then tap or click in the **Card number** field. Depending on the device and 
platform, you may see a chooser showing payment methods stored for the browser, like the one below. 

<figure class="w-figure">
  <img 
    src="images/credit-card-autofill.png" 
    alt="Two screenshots of a payment form in Chrome on an Android phone. One shows the built-in 
    browser payment card selector; the other shows dummy autofilled values.">
  <figcaption class="w-figcaption">Built-in browser payment chooser and autofill.</figcaption>
</figure>

Once you select a payment method and enter your security code, the browser autofills the form using 
the payment card `autocomplete` values you added to the form:

* `cc-number`
* `cc-name`
* `cc-exp`
* `cc-csc`

Many browsers also check and confirm the validity of credit card numbers and security codes.

{% Aside %}
Don't be alarmed! Your browser may be able to autofill the form with stored payment card data, but 
no payment can be made, and no data is transferred or saved.
{% endAside %}

On a mobile device you'll also notice that you get a numeric keyboard as soon as you tap into the 
**Card number** field. That's because you used `inputmode="numeric"`. For numeric fields this makes 
it easier to enter numbers and impossible to enter non-numeric characters, and nudges users to 
remember the type of data they're entering.

```